lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -193,11 +256,34 @@ module.exports.prototype = {
                 var openingBrace = file.getFirstNodeToken(node);
                 var prevToken = file.getPrevToken(openingBrace);
 
-                errors.assert.sameLine({
-                    token: prevToken,
-                    nextToken: openingBrace,
-                    message: 'Newline before curly brace for block statement is disallowed'
-                });
+                if (hasMultiLineEx === true) {
+                    var parentNode = node.parentNode;
+                    var parentNextToken = file.getFirstNodeToken(parentNode);
+                    var openingRoundBrace = file.findNextToken(parentNextToken, 'Punctuator', '(');
+
+                    var closingRoundBrace = file.findPrevToken(openingBrace, 'Punctuator', ')');
+
+                    if (openingRoundBrace && closingRoundBrace &&
```

Comment:
There is a lot of nested `if` which usually not a good sign. Would you mind trying [requireEarlyReturn](http://jscs.info/rule/requireEarlyReturn)?